File: Editor/QuestEditor/Commands/AddStage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using QuestEditor.Models;
using QuestEditor.Proxy;

namespace QuestEditor.Commands
{
    public class AddStage : Common.ICommand
    {
        ProxyQuest mQuest;
        ProxyStage mStage;

        public AddStage(ProxyQuest quest, ProxyStage stage)
        {
            mQuest = quest;
            mStage = stage;
        }

        public virtual void Execute()
        {
            mQuest.AddStage(mStage);
        }

        public virtual void Unexecute()
        {
            mQuest.RemoveStage(mStage);
        }

        public Boolean IsReversible
        {
            get
            {
                return true;
            }
        }
    }
}

File: Editor/LevelEditor/Commands/AddStage.cpp
#include "Stdafx.h"
#include "..\Proxy\QuestProxy.h"
#include "..\Proxy\StageProxy.h"
#include "AddStage.h"

namespace LevelEditor
{
	namespace Commands
	{


		AddStage::AddStage( Proxy::QuestProxy^ quest, Proxy::StageProxy^ stage )
		{
			mQuest = quest;
			mStage = stage;
		}

		void AddStage::Execute()
		{
			mQuest->AddStage(mStage);
		}

		void AddStage::Unexecute()
		{
			mQuest->RemoveStage(mStage);
		}

	}
}